client/homescreen/index.tsx

```diff
@@ -8,37 +8,46 @@ import { getSetting } from '@woocommerce/wc-admin-settings';
 import {
 	ONBOARDING_STORE_NAME,
 	withOnboardingHydration,
+	WCDataSelector,
 } from '@woocommerce/data';
 import { getHistory, getNewPath } from '@woocommerce/navigation';
-
+import type { History } from 'history';
 /**
  * Internal dependencies
  */
 import Layout from './layout';
 
-const Homescreen = ( { profileItems, query } ) => {
+type HomescreenProps = ReturnType< typeof withSelectHandler > & {
+	query: Record< string, string >;
+};
+
+const Homescreen = ( { profileItems, query }: HomescreenProps ) => {
 	const { completed: profilerCompleted, skipped: profilerSkipped } =
 		profileItems || {};
 	if ( ! profilerCompleted && ! profilerSkipped ) {
-		getHistory().push( getNewPath( {}, '/setup-wizard', {} ) );
+		( getHistory() as History ).push(
+			getNewPath( {}, '/setup-wizard', {} )
+		);
 	}
 
 	return <Layout query={ query } />;
 };
 
 const onboardingData = getSetting( 'onboarding', {} );
 
+const withSelectHandler = ( select: WCDataSelector ) => {
+	const { getProfileItems } = select( ONBOARDING_STORE_NAME );
+	const profileItems = getProfileItems();
```

Comment:
It doesn't look like much but check this out in VSCode, you'll see that all the typing for profileItems is there. 🎉